platform/mellanox/mlnx-platform-api/sonic_platform/dpuctl_hwm.py

```diff
@@ -0,0 +1,236 @@
+
+    def __enter__(self):
+        self.file_obj = open(self.file_name, 'w', encoding="utf-8")
+        return self.file_obj
+
+    def __exit__(self, *args):
+        self.file_obj.close()
+
+
+class DpuCtlPlat():
+    """Class for Per DPU API Call"""
+    def __init__(self, dpu_index):
+        self.index = dpu_index + 1
+        self._name = f"dpu{self.index}"
+        self.set_go_down_path = os.path.join(SYSTEM_BASE,
+                                             f"dpu{self.index}_rst")
+        self.set_pwr_path = os.path.join(SYSTEM_BASE,
+                                         f"dpu{self.index}_pwr")
+        self.set_pwr_f_path = os.path.join(SYSTEM_BASE,
+                                           f"dpu{self.index}_pwr_force")
+        self.get_dpu_rdy_path = os.path.join(EVENT_BASE,
+                                             f"dpu{self.index}_ready")
+        self.set_dpu_perst_en_path = os.path.join(SYSTEM_BASE,
+                                                  f"dpu{self.index}_perst_en")
+
+    def write_file(self, file_name, content_towrite):
+        """Write given value to file only if file exists"""
+        try:
+            with DataWriter(file_name) as file_obj:
+                file_obj.write(content_towrite)
+        except (ValueError,
+                IOError,
+                PermissionError,
+                FileNotFoundError) as file_write_exc:
+            logger.log_error(f'{self.get_name()}:Failed to write'
+                             f'{content_towrite} to file {file_name}')
+            raise type(file_write_exc)(
+                f"{self.get_name()}:{str(file_write_exc)}")
+        return True
+
+    def get_name(self):
+        """Return name of the DPU"""
+        return self._name
+
+    def dpu_go_down(self):
+        """Per DPU going down API"""
+        get_shtdn_ready_path = os.path.join(EVENT_BASE,
+                                            f"dpu{self.index}_shtdn_ready")
+        try:
+            get_shtdn_inotify = InotifyHelper(get_shtdn_ready_path)
+            dpu_shtdn_rdy = get_shtdn_inotify.add_watch(WAIT_FOR_SHTDN, 1)
+        except (FileNotFoundError, PermissionError) as inotify_exc:
+            raise type(inotify_exc)(f"{self.get_name()}:{str(inotify_exc)}")
+        if dpu_shtdn_rdy is None:
+            print(f"{self.get_name()}: Going Down Unsuccessful")
+            self.dpu_power_off(forced=True)
+            self.dpu_power_on(forced=True)
+            return
+
+    def dpu_power_off(self, forced=False):
+        """Per DPU Power off API"""
+        print(f"{self.get_name()}: Power off forced={forced}")
+        if forced:
+            self.write_file(self.set_pwr_f_path, "1")
+        else:
+            self.write_file(self.set_go_down_path, "1")
+            self.dpu_go_down()
+            self.write_file(self.set_pwr_path, "1")
+        print(f"{self.get_name()}: Power Off complete")
+
+    def dpu_power_on(self, forced=False, count=4):
```

Comment:
Let's split this function into two: 
dpu_power_on(self)
dpu_power_on_force(self, count=4)

It will improve the redability. If the power on fails we should force power on.
Force power on should the force power off

Reply:
Implemented as a private function without changes to external view of `dpu_power_on` function